test/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter_test.cc

```diff
@@ -274,6 +274,31 @@ TEST_F(GrpcJsonTranscoderConfigTest, InvalidQueryParameter) {
   EXPECT_FALSE(transcoder);
 }
 
+TEST_F(GrpcJsonTranscoderConfigTest, DecodedQueryParameterWithEncodedJsonName) {
+  JsonTranscoderConfig config(
+      getProtoConfig(TestEnvironment::runfilesPath("test/proto/bookstore.descriptor"),
+                     "bookstore.Bookstore"),
+      *api_);
+
+  // When "json_name" is percent encoded, but the field name in query parameter
+  // is percent decoded, it will not match, transcoding fails.
+  // * json_name = "search%5Bencoded%5D", defined in "test/proto/bookstore.proto".
+  // " the query parameter is "search[encode]".
+  Http::TestRequestHeaderMapImpl headers{{":method", "POST"},
+                                         {":path", "/shelf?shelf.search[encode]=Google"}};
```

Comment:
```suggestion
                                         {":path", "/shelf?shelf.search[encoded]=Google"}};
```
otherwise it doesn't match test proto

Reply:
Done